src/screens/Customers/AddPayment.tsx

```diff
@@ -0,0 +1,99 @@
+import { EMPTY_PAYMENT } from '../../lib/redux/customerDataSlice';
+import { selectCurrentUserId } from '../../lib/redux/userData';
+import { createPaymentAndUpdateCustomerBalance } from '../../lib/airtable/request';
+import moment from 'moment';
+
+const styles = (theme: Theme) =>
+    createStyles({
+        content: {
+            color: theme.palette.text.primary,
+        },
+        amountOwedContainer: {
+            display: 'flex',
+            flexDirection: 'column',
+            border: `1px solid ${theme.palette.text.secondary}`,
+            padding: '10px',
+            borderRadius: '6px',
+            marginBottom: '20px',
+        },
+        outlined: {
+            padding: '15px 15px 10px',
+            border: `1px solid ${theme.palette.text.secondary}`,
+            borderRadius: '6px',
+            marginTop: '10px',
+        },
+    });
+
+interface AddPaymentProps extends RouteComponentProps {
+    classes: { content: string; outlined: string; amountOwedContainer: string; };
+}
+
+
+function AddPayment(props: AddPaymentProps) {
+    const { classes } = props;
+    const history = useHistory();
+
+    const currentCustomer: CustomerRecord | undefined = useSelector(selectCurrentCustomer);
+    const currentUserId: string | undefined = useSelector(selectCurrentUserId);
+    const [paymentAmount, setPaymentAmount] = useState(0.0);
+    const [loading, setLoading] = useState(false);
+
+    if (!currentCustomer || !currentUserId) {
+        return <Redirect to={'/customers'} />
+    }
+
+    const handleSetPaymentAmountInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+        setPaymentAmount(parseFloat(event.target.value as string) || 0);
+    }
+
+    // TODO: Error handling
+    // TODO: making paymentAmount a required field
+    // TODO: make this a controlled component
+    const handleSubmit = (event: React.MouseEvent) => {
+        // Prevent page refresh on submit
+        event.preventDefault();
+        setLoading(true);
+
+        const payment = JSON.parse(JSON.stringify(EMPTY_PAYMENT));
+        payment.amount = paymentAmount;
+        payment.date = moment().toISOString();
+        payment.billedToId = currentCustomer.id;
+        payment.collectedById = currentUserId;
+
+        createPaymentAndUpdateCustomerBalance(payment, currentCustomer).then(history.goBack);
+    }
+
+    const totalAmountOwed = currentCustomer.totalAmountBilledfromInvoices - currentCustomer.totalAmountPaidfromPayments;
+
+    return (
+        <BaseScreen title="Add Payment" leftIcon="backNav">
+            <div className={classes.content}>
+                <div className={classes.amountOwedContainer}>
+                    <Typography variant='body1' gutterBottom>
+                        Remaining Balance
+                    </Typography>
+                    <Typography variant='h1'>
+                        {totalAmountOwed} Ks
+                    </Typography>
+                </div>
```

Comment:
I feel like this can be replaced by the `OutlinedCardList` component with some customization

Reply:
^ agreed, also the different naming of Remaining Balance vs Amount Owed is a little confusing (though that's kind of nitpicky)

Reply:
`remainingBalance` is a better name, thanks!